internal/ais/combinemd.go

```diff
@@ -0,0 +1,138 @@
+package ais
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/antchfx/xmlquery"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/fsutil"
+)
+
+const CombineMDActivityName = "combine-metadata-files"
+
+type (
+	CombineMDActivity       struct{}
+	CombineMDActivityParams struct {
+		AreldaRelPath string
+		METSRelPath   string
+		WorkingDir    string
```

Comment:
Maybe call it `localDir` like in the workflow, `workingDir` (from the config) would be the parent directory.

Reply:
Feel free to ignore.

Reply:
Working dir is the config value, used by the worker for all workflows. Local dir is the specific one created for this workflow. Definitely not clear.

Reply:
Hmm, looking at this again the name of `localDir` is deterministic and will be the same for multiple workflow runs if the workingDir is the same (which is always true for a single worker), and the AIP UUID is the same.  If two workflows got started with the same UUID it could cause file lock or race conditions.

It's probably not a problem right now because we are limiting the number of concurrent running workflows to one, but if we ever do increase the number of workflows the worker can simultaneously process it could be an issue.

Reply:
That's a great point and something worth looking at in detail, even if you deploy two workers using the same working directory.

When we [start these workflows](https://github.com/artefactual-sdps/enduro/blob/main/internal/workflow/processing.go#L1136-L1141) from Enduro we also use the AIP UUID for the workflow ID, but we don't set a [`WorkflowIDReusePolicy`](https://pkg.go.dev/go.temporal.io/api/enums/v1#WorkflowIdReusePolicy) and I'm not sure what would happen if unspecified. This could be configurable in Enduro, but we still need to consider all options in the workflow.